### PR TITLE
add RP binary info

### PR DESCRIPTION
**Pull request by Georges760**

### CODE REVIEW

File: firmware/Cargo.toml [hunk 1]
```diff
@@ -89,11 +91,16 @@ keyberon = { git = "https://github.com/TeXitoi/keyberon", version = "0.2.0" }
 phf = { version = "0.11.2", default-features = false }
 packed_struct = { version = "0.10.1", default-features = false }
 rand = { version = "0.8.5", default-features = false, features = ["small_rng"] }
+rp-binary-info = { git = "https://github.com/rp-rs/rp-binary-info.git", commit = "141069dd1685885d35ce2eaa3b345b19288f56e4", optional = true}
```
Comment (simmsb) [inline firmware/Cargo.toml]:
```suggestion
rp-binary-info = { git = "https://github.com/rp-rs/rp-binary-info.git", optional = true}
```
Cargo doesn't use `commit` but `rev`, but I don't think we need to lock the commit explicitly

File: firmware/Cargo.toml [hunk 2]
```diff
@@ -89,11 +91,16 @@ keyberon = { git = "https://github.com/TeXitoi/keyberon", version = "0.2.0" }
...
 
 [features]
 probe = ["defmt", "defmt-rtt", "panic-probe", "shared/defmt", "embassy-embedded-hal/defmt", "embassy-sync/defmt", "embassy-executor/defmt", "embassy-time/defmt", "embassy-time/defmt-timestamp-uptime", "embassy-rp/defmt", "embassy-usb/defmt", "postcard/use-defmt"]
 bootloader = ["embassy-boot-rp"]
+binaryinfo = ["rp-binary-info"]
 
 m2 = []
 
 default = []
```
Comment (simmsb) [inline firmware/Cargo.toml]:
```suggestion
default = ["binaryinfo"]
```

I'm happy to have binary info on by default

File: firmware/src/binary_info.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    concat!(
+        "mod taps, layers, chords, mouse keys, cirque trackpad, neopixel animations, single firmware binary",
+        "\0"
+    ),
+);
+static PROGRAM_BUILD_ATTRIBUTE: IdAndString = custom_string(
+    TAG_RASPBERRY_PI,
+    ID_RP_PROGRAM_BUILD_ATTRIBUTE,
+    concat!(
+        include!(concat!(env!("OUT_DIR"), "/build_attribute.txt")),
```
Comment (simmsb) [inline firmware/src/binary_info.rs]:
```suggestion
        include_str!(concat!(env!("OUT_DIR"), "/build_attribute.txt")),
```

`include!` pastes the file into the file as source code.

**Review by simmsb — CHANGES_REQUESTED** [review]
Thanks for adding this! Just a few changes

File: firmware/src/binary_info.rs (hunk 3 above)
Comment (Georges760) [reply firmware/src/binary_info.rs]: i did fix the literral issue with same solution as build_date, but your solution is better => reverted mine and use yours

**Review by simmsb — APPROVED** [review]

**Pull request merged**